app/models/result.rb

```diff
@@ -4,8 +4,7 @@ class Result < ApplicationRecord
 
   before_save :denormalize_department
 
-  belongs_to :assessment, counter_cache: true
-  belongs_to :assignment, required: true
+  belongs_to :assignment
```

Comment:
Should `required: true` remain on this line? Perhaps, we no longer need it as it is the default in Rails 5. I do remember you saying in the past that because we updated from Rails 4, we have to be explicit. It is possible that that only applies for models added before the Rails 5.0 upgrade.